webapp/channels/src/utils/post_utils.ts

```diff
@@ -796,7 +796,38 @@ export function getUserOrGroupFromMentionName(
     groups: Record<string, Group>,
     groupsDisabled?: boolean,
     getMention = getMentionDetails,
+    remoteClusters?: Array<{remote_id: string; name: string; display_name: string}>,
 ): [UserProfile?, Group?] {
+    // Handle special case for remote user mentions in the format username:clustername
+    if (mentionName.includes(':')) {
+        const [username, clusterName] = mentionName.split(':');
+        if (username && clusterName) {
+            // Find remote users with matching username (could be multiple)
+            const matchingRemoteUsers = Object.values(users).filter((user) =>
+                user.username === username && user.remote_id,
+            );
+            if (matchingRemoteUsers.length === 1) {
+                // If we only have one remote user with this username, use it
+                return [matchingRemoteUsers[0], undefined];
+            } else if (matchingRemoteUsers.length > 1 && remoteClusters?.length) {
```

Comment:
nit, nonblocking: this may lead to bad highlights in certain cases (although very rare). I'd suggest to always create the map to get the cluster from there, and controlling the case in which the cluster doesn't exist for the mention not to be highlighted, but I wouldn't block on this.